cpp/src/aztec/rollup/proofs/join_split/join_split_circuit.cpp

```diff
@@ -254,17 +254,33 @@ join_split_outputs join_split_circuit_component(join_split_inputs const& inputs)
     // A: By passing a signature to the circuit, the 'signing private key' doesn't need to be passed to the proof
     // construction software. This is useful for multisigs, offline signing, etc., so that the proof construction
     // software (or machine) doesn't have access to the signing private key.
-    verify_signature(inputs.public_value.value,
-                     inputs.public_owner,
-                     public_asset_id.value,
-                     output_note_1_commitment,
-                     output_note_2.commitment,
-                     nullifier1,
-                     nullifier2,
-                     signer,
-                     inputs.backward_link,
-                     inputs.allow_chain,
-                     inputs.signature);
+    // Dec 21, 2022: Added exception for merge notes
+    const bool_ct verified = verify_signature(inputs.public_value.value,
+                                              inputs.public_owner,
+                                              public_asset_id.value,
+                                              output_note_1_commitment,
+                                              output_note_2.commitment,
+                                              nullifier1,
+                                              nullifier2,
+                                              signer,
+                                              inputs.backward_link,
+                                              inputs.allow_chain,
+                                              inputs.signature);
+    // Check if we can elide our signature, first we check if this is a note merge,
+    // for the case of spending a bunch of owned notes
+    // This is a merge if one this is a 'send' proof type and one of the output notes has value 0
+    const bool_ct is_merge_send = is_send && (output_note_1_value == 0 || output_note_2_value == 0);
+    // If we are merging our own notes, we don't need to sign
+    // Note: when computing is_same_owner, we rely on input_note_1.owner == input_note_2.owner being checked already
```

Comment:
feedback: get rid of verified_or_merge_with_same_owners_and_amount, is_merge_send already implies the following:
`const bool_ct is_merge_send = is_send && (output_note_1_value == 0 || output_note_2_value == 0) && is_same_owner && is_same_amount;`

then have (verified || is_merge_send).assert_equal....